bob-common/src/metrics/exporters/global_exporter/mod.rs

```diff
@@ -0,0 +1,80 @@
+use metrics::{GaugeValue, Key, Recorder};
+
+const NO_RECORDERS_MSG: &str = "\
+at least one recorder is expected\
+(no reason to build and install global\
+recorder without any local exporters)";
+
+mod rate_processor;
+use rate_processor::RateProcessor;
+
+pub(crate) struct GlobalRecorder {
+    recorders: Vec<Box<dyn Recorder>>,
+    rate_processor: RateProcessor,
+}
+
+impl GlobalRecorder {
+    pub(crate) fn new(recorders: Vec<Box<dyn Recorder>>) -> Self {
+        assert!(recorders.len() != 0, "{}", NO_RECORDERS_MSG);
+        Self {
+            recorders,
+            rate_processor: RateProcessor::run_task(),
+        }
+    }
+}
+
+// NOTE: first recorder is processed separately to avoid redundant clone (value is moved)
+impl Recorder for GlobalRecorder {
+    fn register_gauge(
```

Comment:
You have 3 almost identical fns, maybe it better to do smth like this?
```
fn  register_with(&self, f: fn(...args), ...args) {
    for rec in self.recorders.iter() {
        f(&rec, ...args);
    }
} 
```

Reply:
It's trait implementation, and trait's structure is defined in `metrics` libs.
I don't understand you, you wan't to make register_`metric` functions to be wrappers above register_with or what?

Reply:
Yes, you do the same operation 3 times but with different metrics. But maybe it unnecessary code complication